js/stdlib/ts/ALGO.ts

```diff
@@ -1199,6 +1199,34 @@ const getApplicationInfoM = async (idn:BigNumber): Promise<OrExn<AppInfo>> => {
   return 'exn' in res ? res : { val: res.val.application };
 };
 
+const getDeletedApplicationInfoM = async (idn: BigNumber): Promise<OrExn<AppInfo>> => {
+  const dhead = 'getDeletedApplicationInfoM'
+  const indexer = await getIndexer();
+  const query = indexer.searchForTransactions()
+                       .txType('appl')
+                       .applicationID(bigNumberToNumber(idn))
+                       .limit(1);
```

Comment:
First, I don't believe that there is a guarantee you will get the first one.

Second, you should cast this to the correct `ApiCall` type so that the type checker works

Reply:
Following the transaction querying logic in the indexer, you come to a line of the relevant query gen (see link) that shows this query will always get the first txn. However, I agree that since this is relying on an implementation detail of the indexer it's a good idea to add a sanity check 
 https://github.com/algorand/indexer/blob/82b9fe3e8e0cd5599e219a5adff79d30119099ea/idb/postgres/postgres.go#L825-L828